### PR TITLE
Unit test for ListAtContext

### PR DESCRIPTION
**Pull request by bquantump**

### CODE REVIEW

File: ClientTests/ResourceListOperationsTest.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace azure_proto_core_test
+{
+    public class ResourceListOperationsTest
+    {
+        [TestCase]
+        public void TestArmResonseArmResource()
+        {
+             TestListActivator<ArmResourceOperations, ArmResource>();
+        }
+        private static void TestListActivator<T, U>()
```
Comment (m-nash) [inline ClientTests/ResourceListOperationsTest.cs]: Can we give the type parameters semantic names like TOperations etc?

File: ClientTests/ResourceListOperationsTest.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            var testMethod = typeof(azure_proto_core.ResourceListOperations).GetMethod("CreateResourceConvertor", BindingFlags.Static | BindingFlags.NonPublic);
+            var asGeneric = testMethod.MakeGenericMethod(new System.Type[] {typeof(T), typeof(U)});
+            var context = new ArmClientContext(new Uri("https://management.azure.com"), new DefaultAzureCredential());
+            var function =  (Func<GenericResourceExpanded, T>) asGeneric.Invoke(null, new object[] {context, new ArmClientOptions()} );
+            var resource = new GenericResourceExpanded();
+            resource.Location = "East US";
+            resource.Tags =  new Dictionary<string, string>{ {"tag1", "value1"} };
+            function.DynamicInvoke(new object [] {resource});
```
Comment (m-nash) [inline ClientTests/ResourceListOperationsTest.cs]: Can we trap the return type here which I believe should be a T and assert some of its properties.  Like it should have the same context and resource property values that we set on lines 23 and 25-27?

File: azure-proto-compute/Placeholder/AvailabilitySetData.cs [hunk 3]
```diff
@@ -17,6 +17,8 @@ public AvailabilitySetData(Azure.ResourceManager.Compute.Models.AvailabilitySet
             }
         }
 
+
```
Comment (m-nash) [inline azure-proto-compute/Placeholder/AvailabilitySetData.cs:20]: I think you can revert this file

File: azure-proto-core/ResourceListOperations.cs [hunk 4]
```diff
@@ -196,10 +191,17 @@ private static void Validate(ResourceIdentifier id)
         {
             return new PhWrappingAsyncPageable<GenericResourceExpanded, TOperations>(
                 result,
-                s => Activator.CreateInstance(
+                CreateResourceConvertor<TOperations, TResource>(clientContext, clientOptions));
+        }
+
+        private static Func<GenericResourceExpanded, TOperations> CreateResourceConvertor<TOperations, TResource>(ArmClientContext clientContext, ArmClientOptions clientOptions)
+        where TOperations : ResourceOperationsBase<TOperations, TResource>
```
Comment (m-nash) [inline azure-proto-core/ResourceListOperations.cs:198]: where clauses should be indented similar to the other methods

**Review by m-nash — CHANGES_REQUESTED** [review]

**Review by markcowl — CHANGES_REQUESTED** [review]
I think this is fine, modulo the spelling change, and I don't want to spend too much time on this, as we won't need this once we move to the .Net repo, but I wonder if this wouldn't be simpler if we simply had a constructor inthe concrete Operatiosn classes that took the model type from the generated code as a parameter

**Review by m-nash — APPROVED** [review]

**Pull request merged**